gulpfile.js

```diff
@@ -416,3 +418,46 @@ gulp.task('test-e2e', function() {
     'stop-webserver'
   );
 });
+
+/**
+ * Deploys production-optimized app build to the "origin/gh-branch". Every new
+ * deployment comes with a separate commit attributed with the date and source
+ * branch revision. If tree has some changes that are not committed yet, there
+ * will be a warning, but all local changes will be deployed anyway.
+ */
+gulp.task('deploy', ['build-production'], function() {
+  const wrapIntoPromise = (method, args) => new Promise((resolve, reject) => {
+    git[method].call(
+      git, args, (err, result) => err ? reject(err) : resolve(result)
+    );
+  });
+
+  return Promise
+    .all([
+      wrapIntoPromise('revParse', { args: '--short HEAD', quiet: true }),
```

Comment:
This two git-related steps (git revision and warning about local changes) are not necessary, they bring a bit of complexity, but I found them useful. Anyway I'm not 100% convinced that super-duper needed, so let me know if you prefer to get rid of it.